frontend/controller/app/chatroom.js

```diff
@@ -14,7 +14,22 @@ sbp('okTurtles.events/on', JOINED_CHATROOM, ({ identityContractID, groupContract
   const rootState = sbp('state/vuex/state')
   if (rootState.loggedIn?.identityContractID !== identityContractID) return
   if (!rootState.chatroom.currentChatRoomIDs[groupContractID] || rootState.chatroom.pendingChatRoomIDs[groupContractID] === chatRoomID) {
-    sbp('state/vuex/commit', 'setCurrentChatRoomId', { groupID: groupContractID, chatRoomID })
+    let attemptCount = 0
+    // Sometimes, the state may not be ready (it needs to be copied from the SW
+    // to Vuex). In this case, we try again after a short delay.
+    // TODO: Figure out a better way of doing this that doesn't require a timeout
```

Comment:
Can you expand this comment so that someone who is revisiting this code in 1 month understands what the issues are in order to implement this TODO? i.e. answering the question of why the `JOINED_CHATROOM` event can't simply be called after the state is copied over, and the relationship between `EVENT_HANDLED` and this `JOINED_CHATROOM` event?